### DISCUSSION TITLE
Add delete method for cookies

### DISCUSSION THREAD

**Discussion by mikeybinns** [remix-run/remix#6934] [Proposals]
I think we should add a delete method to cookies which just acts as a shortcut for serialising a cookie with an expires option set in the past.
```ts
await cookie.delete()
// is the same as
await cookie.serialize(null, {
  expires: new Date(1970, 0, 1),
})
```
This is easier to remember and easier to write and read for developers.
It would accept no value or options because if you're deleting the cookie, they don't matter.

**Comment by wong2** [comment]
May I ask why are you using `expires` instead of `maxAge: -1`?

**Reply by mikeybinns** [reply]
No particular reason, does the same thing, it's just a different method.

**Reply by wong2** [reply]
Thanks!

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
